server/src/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java

```diff
@@ -639,10 +648,57 @@ else if (jobResult instanceof Throwable)
                     throw new RuntimeException("Unexpected exception", (Throwable)jobResult);
             }
 
+            updateUserVmServiceOffering(userVm, vmSnapshotVo);
```

Comment:
No need to update service offering here, the else part is just to create an entry in the DB for the revert to snapshot job.